lib/internal/process/promises.js

```diff
@@ -26,6 +33,7 @@ function setupPromises(scheduleMicrotasks) {
     if (hasBeenNotified !== undefined) {
       hasBeenNotifiedProperty.delete(promise);
       if (hasBeenNotified === true) {
+        promiseInternals.untrackPromise(promise);
```

Comment:
Right, it only makes it for ones that are not handled before the event, and only if the event does not have listeners.